### PR TITLE
[WIP] Debugging Irregular Tile Size

### PR DESCRIPTION
**Pull request by zjing14**
- Solved an issue in ThreadwiseCopy
- Added irregular tiling size
- Added M/N padding

### CODE REVIEW

File: device_operation/include/device_gemm_xdl.hpp [hunk 1]
```diff
@@ -78,10 +78,14 @@ struct DeviceGemmXdl
...
 
+        const auto PadM = (MPerBlock - M % MPerBlock) % MPerBlock;
+
+        std::cout << "PadM = " << PadM << " M = " << M + PadM << std::endl;
+
         const auto a_grid_desc_k0_m_k1 =
             transform_tensor_descriptor(a_grid_desc_m_k,
                                         make_tuple(make_unmerge_transform(make_tuple(K0, K1Number)),
-                                                   make_pass_through_transform(M)),
+                                                   make_pad_transform(M, I0, PadM)),
```
Comment (asroy) [inline device_operation/include/device_gemm_xdl.hpp:88]: you can use make_right_pad_transform

File: device_operation/include/device_gemm_xdl.hpp [hunk 2]
```diff
@@ -105,10 +109,14 @@ struct DeviceGemmXdl
...
 
+        const auto PadN = (NPerBlock - N % NPerBlock) % NPerBlock;
+
+        std::cout << "PadN = " << PadN << " N = " << N + PadN << std::endl;
+
         const auto b_grid_desc_k0_n_k1 =
             transform_tensor_descriptor(b_grid_desc_k_n,
                                         make_tuple(make_unmerge_transform(make_tuple(K0, K1Number)),
-                                                   make_pass_through_transform(N)),
+                                                   make_pad_transform(N, I0, PadN)),
```
Comment (asroy) [inline device_operation/include/device_gemm_xdl.hpp:119]: Same here

File: device_operation/include/device_gemm_xdl.hpp [hunk 3]
```diff
@@ -117,14 +125,27 @@ struct DeviceGemmXdl
...
+                return make_naive_tensor_descriptor(make_tuple(M, N), make_tuple(I1, StrideC));
+            }
+        }();
+
+        const auto PadM = (MPerBlock - M % MPerBlock) % MPerBlock;
+        const auto PadN = (NPerBlock - N % NPerBlock) % NPerBlock;
+
+        const auto c_grid_desc_m_n_ = transform_tensor_descriptor(
+            c_grid_desc_m_n,
+            make_tuple(make_pad_transform(M, I0, PadM), make_pad_transform(N, I0, PadN)),
```
Comment (asroy) [inline device_operation/include/device_gemm_xdl.hpp:144]: same here

File: host/driver_offline/include/driver_gemm_xdlops_v2r3.hpp [hunk 4]
```diff
@@ -6,6 +6,15 @@
 #include "tensor_descriptor_helper.hpp"
 #include "gridwise_gemm_xdlops_v2r3.hpp"
 
+struct OpPassThrough
```
Comment (asroy) [inline host/driver_offline/include/driver_gemm_xdlops_v2r3.hpp:9]: It's  already defined under device_operation/element_wise_operation.hpp

File: composable_kernel/include/tensor_operation/threadwise_tensor_slice_transfer.hpp [hunk 5]
```diff
@@ -165,7 +165,7 @@ struct ThreadwiseTensorSliceTransfer_v1r3
                 static_for<1, nDim, 1>{}([&](auto i) {
                     index_t tmp = ordered_access_idx[I0];
 
-                    static_for<0, i, 1>{}([&](auto j) {
+                    static_for<1, i, 1>{}([&](auto j) {
```
Comment (asroy) [inline composable_kernel/include/tensor_operation/threadwise_tensor_slice_transfer.hpp:168]: Nice catch!

**Comment by asroy** [comment]
This PR fixed some bugs in CK. Have those fixes gone into CK?

**Comment by zjing14** [comment]
> This PR fixed some bugs in CK. Have those fixes gone into CK?

Yes. All fixes are in develop or other PRs.

**Comment by asroy** [comment]
Do we still need irregular tile sizes?

**Pull request closed without merging**